segment.go

```diff
@@ -90,17 +90,21 @@ func (seg *segment) set(key, value []byte, hashVal uint64, expireSeconds int) (e
 		hdr.slotId = slotId
 		hdr.hash16 = hash16
 		hdr.keyLen = uint16(len(key))
+		originAccessTime := hdr.accessTime
 		hdr.accessTime = now
 		hdr.expireAt = expireAt
 		hdr.valLen = uint32(len(value))
 		if hdr.valCap >= hdr.valLen {
 			//in place overwrite
-			seg.totalTime += int64(hdr.accessTime) - int64(now)
+			seg.totalTime += int64(hdr.accessTime) - int64(originAccessTime)
 			seg.rb.WriteAt(hdrBuf[:], matchedPtr.offset)
 			seg.rb.WriteAt(value, matchedPtr.offset+ENTRY_HDR_SIZE+int64(hdr.keyLen))
 			seg.overwrites++
 			return
 		}
+		// avoid unnecessary memory copy.
+		seg.delEntryPtr(slotId, hash16, seg.slotsData[idx].offset)
```

Comment:
When the newVal is larger than original, segment will write newVal to new space. The original space maybe not expired and doesn't marked deleted. So the original space will copy to the end of ringbuf when evacuating space.